tests-e2e/cypress/integration/permissions_spec.ts

```diff
@@ -0,0 +1,229 @@
+        // # Create a new direct message and visit it.
+        cy.visitNewDirectMessage('user-1', 'anne.stone').then((channel: Channel) => {
+            // # Run the /export slash command.
+            cy.exportSlashCommand();
+
+            // # Open the DM with the @channelexport bot.
+            cy.visitDMWithBot();
+
+            // * Verify that there is a file to be downloaded in the bot's
+            // message.
+            cy.verifyFileCanBeDownloaded(channel.name);
+        });
+    });
+
+    it('ID 13 - User can export a direct message channel with self', () => {
+        // # Get the user from the username to retrieve its ID.
+        cy.apiGetUserByUsername('user-1').then((user: UserProfile) => {
+            // # Visit the direct message channel with self.
+            cy.visit('/ad-1/messages/@user-1');
+
+            // # Run the /export slash command.
+            cy.exportSlashCommand();
+
+            // # Open the DM with the @channelexport bot.
+            cy.visitDMWithBot();
+
+            // * Verify that there is a file to be downloaded in the bot's
+            // message.
+            cy.verifyFileCanBeDownloaded(`${user.id}__${user.id}`);
+        });
+    });
+
+    it('ID 14 - User can export a bot message channel', () => {
+        // # Open the DM with the @channelexport bot.
+        cy.visitDMWithBot();
+
+        // # Run the /export slash command.
+        cy.exportSlashCommand();
+
+        // # Get the user from the username to retrieve its ID.
+        cy.apiGetUserByUsername('user-1').then((user: UserProfile) => {
+            // # Get the bot from its username to retrieve its ID.
+            cy.apiGetUserByUsername('channelexport').then((bot: UserProfile) => {
+                // * Verify that there is a file to be downloaded in the bot's
+                // message.
+                cy.verifyFileCanBeDownloaded(`${user.id}__${bot.id}`);
+            });
+        });
+    });
+
+    it('ID 15 - User can export archived channel', () => {
+        // # Create a new public channel and visit it.
+        cy.visitNewPublicChannel().then((channel: Channel) => {
+            // # Archive the current channel
+            cy.archiveCurrentChannel();
+
+            // * Verify that the channel can be exported via the plugin's API.
+            cy.apiExportChannel(channel.id).then((fileContents: string) => {
+                expect(fileContents).to.contain(fileHeader);
+            });
+        });
+    });
+
+    it('ID 16 - User can export an unarchived channel', () => {
+        // # Create a new public channel and visit it.
+        cy.visitNewPublicChannel().then((channel: Channel) => {
+            // # Archive the current channel
+            cy.archiveCurrentChannel();
+
+            // # Unarchive the channel logged in as an admin
+            cy.apiLogin('sysadmin').then(() => {
+                cy.visit(`/ad-1/channels/${channel.name}`);
```

Comment:
Maybe add as assertion here that the expected channel has loaded:
                + cy.get('#channelHeaderTitle').contains('Channel Export');

Reply:
Thank you! Done in https://github.com/mattermost/mattermost-plugin-channel-export/pull/14/commits/f216df64c0568dd114c524cc4df7b672dd3d7b4a.

Reply:
I'm still seeing the failing test here. After visiting to `channel.name`, it immediately redirects to `town-square`. In my case, it requires to enable "Allow users to view archived channels".

Reply:
Oh I see, that makes sense! It's what Prapti said about the feature not being enabled. It should be fixed with https://github.com/mattermost/mattermost-plugin-channel-export/pull/14/commits/52166724a9f1b4a3da9e26495637cb0a5b442e12!